fireworks/core/launchpad.py

```diff
@@ -1082,6 +1082,26 @@ def resume_fw(self, fw_id):
             self._refresh_wf(fw_id)
         return f
 
+    def checkpoint_fw(self, fw_id):
+        """
+        Given the firework id, mark firework as checkpointed and refresh the workflow
+
+        Args:
+            fw_id(int): firework id
+        """
+        allowed_states = ['RUNNING']
+        f = self.fireworks.find_one_and_update(
```

Comment:
@computron had some discussion here that maybe the state should be updated in the `launches`